packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -145,17 +207,111 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
   Color get barrierColor => null;
 
   Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> forwardAnimation) {
-    return new _PopupMenu(route: this);
+    int selectedIndex = -1;
+    if (initialValue != null) {
```

Comment:
I agree that it's possible, when we have scrolling popup menus, for lists to have thousands of items.  I'll file an issue about addressing this.